### PR TITLE
JP-1780: Clean up jwst.datamodels tests

### LINKED ISSUE #5470: Clean up jwst.datamodels tests that are no longer needed
_Issue [JP-1780](https://jira.stsci.edu/browse/JP-1780) was created on JIRA by [James Davies](https://jira.stsci.edu/secure/ViewProfile.jspa?name=jdavies):_

Since #5433 moved the core `jwst.datamodels` library functionality to `stdatamodels`, there is now a number of tests in `jwst.datamodels` that are actually testing code in `stdatamodels`.  So it would be good to clean up the tests.

- Remove tests that were ported over to `stdatamodels` and exist there.  I.e. the test doesn't reduce coverage here.
- Retain tests that test dependent features of `stdatamodels` by testing with `jwst.datamodles`-specific models or code.  I.e. if the test increases coverage here, keep it.

### PR DESCRIPTION
**Pull request by jdavies-st**
Most of these tests are already ported over to `stdatamodels`.  This removes them and cleans up the remainder with the goal of not dropping test coverage.

Also, adds back in `jwst.datamodels.validation.ValidationWarning`, even though that is now pulled from `stdatamodels.validation`.  This is for backwards compatibility with CRDS (and possibly other packages).

Resolves #5470 / [JP-1780](https://jira.stsci.edu/browse/JP-1780)

### CODE REVIEW

**Review by kmacdonald-stsci — APPROVED** [review]
It all looks good to me.

File: jwst/datamodels/tests/test_schema.py [hunk 1]
```diff
@@ -1,73 +1,39 @@
...
-
-def teardown():
-    shutil.rmtree(TMP_DIR)
-
-
-def test_choice():
-    with pytest.raises(jsonschema.ValidationError):
-        with JwstDataModel(FITS_FILE, strict_validation=True) as dm:
-            assert dm.meta.instrument.name == 'MIRI'
+def test_strict_validation_enum():
```
Comment (eslavich) [inline jwst/datamodels/tests/test_schema.py:20]: Would it be reasonable to dissolve this file and divided its tests among other jwst modules and stdatamodels?  Here's a sketch of where we might put them:

**stdatamodels tests/test_validation.py**
test_strict_validation_enum
test_strict_validation_type
test_strict_validation_date

**jwst jwst/datamodels/tests/test_models.py**
test_mask_model
test_table_size_zero

**jwst, maybe a file specific to testing the asdf extension?**
test_datamodel_schema_entry_points

**jwst, maybe a test_multislit.py?**
test_multislit_model
test_copy_multslit
test_multislit_move_from_fits
test_multislit_append_string
test_all_datamodels_init

**???**
test_data_array -- not sure what this is about

**possible duplicates**
test_list and test_list2 -- we have some tests in stdatamodels for these already
test_invalid_fits -- there's good coverage of PASS_INVALID_VALUES and STRICT_VALIDATION in stdatamodels

File: jwst/datamodels/tests/test_models.py [hunk 2]
```diff
@@ -377,24 +123,19 @@ def test_multislit():
...
-
 def test_multislit_from_image():
     with ImageModel((64, 64)) as im:
         with MultiSlitModel(im) as ms:
             assert len(ms.slits) == 1
             assert ms.slits[0].data.shape == (64, 64)
 
 
-def test_multislit_from_fits_image():
+def test_multislit_from_saved_image(tmp_path):
```
Comment (eslavich) [inline jwst/datamodels/tests/test_models.py:133]: These could go in a multislit-specific test module as well

**Review by eslavich — COMMENTED** [review]
These changes look good, just the one thing about possibly dispersing test_schema.py

File: jwst/datamodels/tests/test_schema.py (hunk 1 above)
Comment (jdavies-st) [reply jwst/datamodels/tests/test_schema.py]: Excellent.  This is done.  I'm not sure what `test_data_array` does either.  Broadcasting?  It's very odd and seems to be doing lots of things.  For now I'm leaving it where it is.

File: jwst/datamodels/tests/test_models.py (hunk 2 above)
Comment (jdavies-st) [reply jwst/datamodels/tests/test_models.py]: Done.

**Pull request merged**